test/lib/commands/completion.js

```diff
@@ -199,47 +196,47 @@ t.test('dump script exits correctly when EPIPE is emitted on stdout', t => {
     })
   }
 
-  completion.exec({}, (err) => {
-    if (err)
-      throw err
-
-    t.equal(data, completionScript, 'wrote the completion script')
-    t.end()
-  })
+  await completion.exec({})
+  t.equal(data, completionScript, 'wrote the completion script')
 })
 
-t.test('non EPIPE errors cause failures', t => {
-  const _write = process.stdout.write
-  const _on = process.stdout.on
-  t.teardown(() => {
-    process.stdout.write = _write
-    process.stdout.on = _on
-  })
-
-  let errorHandler
-  process.stdout.on = (event, handler) => {
-    errorHandler = handler
-    process.stdout.on = _on
-  }
-
-  let data
-  process.stdout.write = (chunk, callback) => {
-    data = chunk
-    process.stdout.write = _write
-    process.nextTick(() => {
-      errorHandler({ errno: 'ESOMETHINGELSE' })
-      callback()
-    })
-  }
-
-  completion.exec({}, (err) => {
-    t.equal(err.errno, 'ESOMETHINGELSE', 'propagated error')
-    t.equal(data, completionScript, 'wrote the completion script')
-    t.end()
-  })
-})
-
-t.test('completion completes single command name', t => {
+// This test was only working by coincidence before, when we switch to full
+// async/await the race condition now makes it impossible to test.  The root of
+// the problem is that if we override stdout.write then other things interfere
+// during testing.
```

Comment:
Should this be marked with a `// TODO:` or `// XXX:` comment?

Reply:
I'll mark it in the command itself where the coverage line is ignored and points to this test